examples/eighth/main.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/beatlabs/patron"
+	patronhttp "github.com/beatlabs/patron/component/http"
+	"github.com/beatlabs/patron/log"
+)
+
+func init() {
+	err := os.Setenv("PATRON_LOG_LEVEL", "debug")
+	if err != nil {
+		fmt.Printf("failed to set log level env var: %v", err)
+		os.Exit(1)
+	}
+	err = os.Setenv("PATRON_JAEGER_SAMPLER_PARAM", "1.0")
+	if err != nil {
+		fmt.Printf("failed to set sampler env vars: %v", err)
+		os.Exit(1)
+	}
+}
+
+var middlewareCors = func(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Add("Access-Control-Allow-Origin", "*")
+		w.Header().Add("Access-Control-Allow-Methods", "GET, POST")
+		w.Header().Add("Access-Control-Allow-Headers", "Origin, Authorization, Content-Type")
+		w.Header().Add("Access-Control-Allow-Credentials", "Allow")
+		h.ServeHTTP(w, r)
+	})
+}
+
+// In the following example, we define a route that serves some random data.
+// We call this route with and without Accept-Encoding headers so we that we test the compression methods
+// $ curl -s localhost:50000/foo | wc -c
+// 1398106
+// $ curl -s localhost:50000/foo -H "Accept-Encoding: nonexisting" | wc -c
+// 1398106
+// $ curl -s localhost:50000/foo -H "Accept-Encoding: gzip" | wc -c
+// 1053068
+// $ curl -s localhost:50000/foo -H "Accept-Encoding: deflate" | wc -c
+// 1053045
+//
+// For ignored routes, we don't see any compression applied, even if we specify a correct header
+// $ curl -s localhost:50000/bar -H "Accept-Encoding: gzip" | wc -c
+// 1398106
+// $ curl -s localhost:50000/bar  | wc -c
+// 1398106
+func main() {
+	name := "eighth"
+	version := "1.0.0"
+
+	err := patron.SetupLogging(name, version)
+	handle(err)
+
+	compressionMiddleware, err := patronhttp.NewCompressionMiddleware().WithIgnoreRoutes("/bar", "/alive", "/ready", "/metrics").Build()
+	handle(err)
+
+	// You could either add the compression middleware per-route, like here ...
+	routesBuilder := patronhttp.NewRoutesBuilder().
+		Append(patronhttp.NewRouteBuilder("/foo", eighth).MethodGet()). //.WithMiddlewares(compressionMiddleware))
+		Append(patronhttp.NewRouteBuilder("/bar", eighth).MethodGet()).
+		Append(patronhttp.NewRouteBuilder("/hello", hello).MethodGet())
+
+	// or pass middlewares to the HTTP component globally, like we do below
+	ctx := context.Background()
+	err = patron.New(name, version).
+		WithRoutesBuilder(routesBuilder).
```

Comment:
I propose the compression middleware to be always created inside patron HTTP component, much like the other middlewares already present e.g. recovery, caching etc.
The only thing that we need is a builder function for the ignore routes in the service builder.
This way the API is minimal and there is no need to pipe into patron the middleware at all...
No need for a builder etc... much simpler and less code.

Reply:
@mantzas Hm, I think you're right, I'll do that. 

Do you think we should also hardcode the compression level for 'deflate' to the maximum value to keep the API minimal? I'd think towards that.

For modern hardware and usual sizes of HTTP transports, the differences are miniscule.    
For larger transport sizes, the speed/size trade-off is more apparent.

Here's some benchmarks I've run
```
size \ deflate |   lvl 1         lvl 9
------------------------------------------
1KiB data      |  0.18ms        0.17ms
1MiB data      |   3.5ms          23ms
1GiB data      |    4.5s           24s
```

Reply:
@tpaschalis could we provide this parameter as a Method in the service builder? This way we can have a sane default and also support configuring the value...